src/react/src/components/SelfServiceDromoUploader.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react';
+import { object, func } from 'prop-types';
+import DromoUploader from 'dromo-uploader-react';
+import MaterialButton from '@material-ui/core/Button';
+import { processDromoResults } from '../util/util';
+
+const uploaderButtonStyle = Object.freeze({
+    backgroundColor: '#62CC74',
+    color: 'white',
+});
+
+const SelfServiceDromoUploader = ({ fileInput, updateFileName }) => {
+    const [isUploaderOpen, setIsUploaderOpen] = useState(false);
+
+    const openUploader = () => setIsUploaderOpen(true);
+    const closeUploader = () => setIsUploaderOpen(false);
+
+    const handleDromoResults = (results, metadata) => {
+        const { filename } = metadata;
+
+        processDromoResults(results, filename, fileInput, updateFileName);
+        closeUploader();
+    };
+
+    return (
+        <>
+            <MaterialButton
+                onClick={openUploader}
+                type="button"
+                variant="contained"
+                style={uploaderButtonStyle}
+                disableRipple
+            >
+                Beta Self Service Upload
+            </MaterialButton>
+
+            <DromoUploader
+                licenseKey="ee427cf5-da27-4f28-a260-c9a17d02ad30"
```

Comment:
As I understand it, the licenseKey is similar to an API key issued specifically for OS Hub and should be secured in some way. Is it restricted to the OS Hub domain name?
Would it work locally with this licenseKey, as well as in all other environments?

Reply:
`licenseKey` is a Dromo Frontend API Key, and it's designed to be embedded in a client app.
It is the same for all environments.

Reply:
Unfortunately, I don't have access to the OSHub Dromo account, so I can't check if there are restricted domains (I suggest there are no). 
But I created my own Dromo account for testing purposes, and there is a description under Whitelisted domains: "Not enforced in development mode."
It allows us to build and test integration locally without restrictions, even if some domain restrictions were enforced.

My suggestion is to leave it as it is.
<img width="706" height="300" alt="Screenshot from 2025-07-18 12-45-20" src="https://github.com/user-attachments/assets/b4b0d1ee-39a2-4421-abbc-e74448e33a8e" />